src/channel_state.ts

```diff
@@ -469,13 +469,20 @@ export class ChannelState<StreamChatGenerics extends ExtendableGenerics = Defaul
 
     // message already exists and not filtered due to timestampChanged, update and return
     if (!timestampChanged && message.id) {
+      let messageIndex;
       if (messageArr[left] && message.id === messageArr[left].id) {
-        messageArr[left] = message;
-        return [...messageArr];
+        messageIndex = left;
       }
 
       if (messageArr[left - 1] && message.id === messageArr[left - 1].id) {
-        messageArr[left - 1] = message;
+        messageIndex = left - 1;
+      }
+
+      if (messageIndex !== undefined) {
+        // own_reactions always be [] in WS events -> ignore for message updates
```

Comment:
I'm not happy with this solution, I think the nice solution would be to extend the `event.message` object with `own_reactions` when the `message.updated` [event is received](https://github.com/GetStream/stream-chat-js/blob/master/src/channel.ts#L1250), but that would require an extra iteration of the `channelState.messages` and wasn't sure how performance-critical that method was

Reply:
Did a second version: https://github.com/GetStream/stream-chat-js/pull/1009 - let me know which one you prefer

Reply:
I like the second one better

Reply:
👍